src/test/run-pass/arbitrary_self_types_raw_pointer_struct.rs

```diff
@@ -8,11 +8,17 @@
 // option. This file may not be copied, modified, or distributed
```

Comment:
Could you also add a test for "complex" cases - i.e. `Rc<*const Self>` & `*const Rc<Self>`?